concurrency/limited_concurrency_singleflight_test.go

```diff
@@ -0,0 +1,133 @@
+package concurrency
+
+import (
+	"context"
+	"strconv"
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"go.uber.org/atomic"
+)
+
+func TestLimitedConcurrencySingleFlight_ForEachNotInFlight_ConcurrencyLimit(t *testing.T) {
+	t.Parallel()
+
+	var (
+		ctx            = context.Background()
+		pool           = NewWorkerPool(10)
+		workersWait    chan struct{}
+		workersToStart sync.WaitGroup
+	)
+
+	forEachNotInFlight := func(tokens []int, f func(context.Context, string) error) {
+		stringTokens := make([]string, len(tokens))
+		for i, token := range tokens {
+			stringTokens[i] = strconv.Itoa(token)
+		}
+		require.NoError(t, pool.ForEachNotInFlight(ctx, stringTokens, f))
+	}
+
+	busyWorker := func(ctx context.Context, s string) error {
+		workersToStart.Done()
+		<-workersWait
+		return nil
+	}
+
+	workersToStart.Add(10)
```

Comment:
This test seems like it will leak goroutines. Should be a cleanup where `workersWait` is closed?